src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1528,6 +1532,14 @@ public static bool HasNames(this TupleExpressionSyntax tuple)
                 return true;
             }
 
+            if (token.IsKind(SyntaxKind.RefKeyword, SyntaxKind.ReadOnlyKeyword) &&
+                token.Parent.IsKind(SyntaxKind.RefType) &&
```

Comment:
didn't you jsut add a new helper that does this?

Reply:
no I didn't... and even if I did, this is a special case because this is used for `var`, which can't be offer in other places after a ref type (eg in a method return type)

Reply:
So this is not just the same as "IsAfterRefTypeContext"?

Reply:
no it's not:
```c#
class C
{
    ref $$
}
```
here you're in a ref type but not in a local declaration and can't use `var`